modules/policy/engine/src/main/java/org/eclipse/sw360/antenna/policy/engine/PolicyViolation.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) Bosch Software Innovations GmbH 2019.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.sw360.antenna.policy.engine;
+
+import java.io.UnsupportedEncodingException;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Base64;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.stream.Collectors;
+
+/**
+ * The result object for a {@link Rule} violation. The {@link PolicyEngine} creates for violations an object of this type
+ * and returns all {@link ThirdPartyArtifact} objects which have caused the policy violation.
+ */
+public class PolicyViolation {
+    private Rule rule;
+    private Collection<ThirdPartyArtifact> failedArtifacts = new HashSet<>();
+
+    PolicyViolation(final Rule rule, final Collection<ThirdPartyArtifact> failingArtifacts) {
+        this.rule = rule;
+        this.failedArtifacts.addAll(failingArtifacts);
+    }
+
+    /**
+     * @return Business id of the associated rule
+     */
+    public String getId() {
+        return rule.getId();
+    }
+
+    /**
+     * @return Business getName of the associated rule
+     */
+    public String getName() {
+        return rule.getName();
+    }
+
+    /**
+     * @return Violation description of the associated rule
+     */
+    public String getDescription() {
+        return rule.getDescription();
+    }
+
+    /**
+     * @return Configured severity of the associated rule
+     */
+    public RuleSeverity getSeverity() {
+        return rule.getSeverity();
+    }
+
+    /**
+     * @return A hash code that uniquely identifies the violation. It is reproducable over different rule evaluations
+     */
+    public String getViolationHash() {
+        String hashBase = failedArtifacts.stream()
+                .map(ThirdPartyArtifact::getPurl)
+                .collect(Collectors.joining(" : ", rule.getId() + " : ", ""));
+        try {
+            MessageDigest messageDigest = MessageDigest.getInstance("MD5");
```

Comment:
Why not just use `hashBase.hashCode()`?

Reply:
hashCode provides only a single int as hash. My impression is, that the likelyhood of collisions is there. The idea of the hash code is, to have a unique id for the violation, to, e.g., use them later in a waiving mechanism. So collisions should be avoided as much as possible.

Reply:
Taking about collisions, how about using SHA1 instead then, also as that's more common in general nowadays? Or do you prefer MD5 as the string is shorter?

Reply:
I think MD5 is good enough and we will never find a collision

Reply:
I used MD5 because the string is shorter, indeed. I compared the results and since this should be in principle usable to refer to the violation, I thought that MD5 is the best compromise.